hashednode.go

```diff
@@ -30,69 +30,48 @@ import (
 	"fmt"
 )
 
-type HashedNode struct {
-	commitment  []byte
-	cachedPoint *Point
-}
+type HashedNode struct{}
 
-func (*HashedNode) Insert([]byte, []byte, NodeResolverFn) error {
+func (HashedNode) Insert([]byte, []byte, NodeResolverFn) error {
 	return errInsertIntoHash
 }
 
-func (*HashedNode) Delete([]byte, NodeResolverFn) (bool, error) {
+func (HashedNode) Delete([]byte, NodeResolverFn) (bool, error) {
 	return false, errors.New("cant delete a hashed node in-place")
 }
 
-func (*HashedNode) Get([]byte, NodeResolverFn) ([]byte, error) {
+func (HashedNode) Get([]byte, NodeResolverFn) ([]byte, error) {
 	return nil, errors.New("can not read from a hash node")
 }
 
-func (n *HashedNode) Commit() *Point {
-	if n.commitment == nil {
-		panic("nil commitment")
-	}
-	if n.cachedPoint == nil {
-		n.cachedPoint = new(Point)
-		n.cachedPoint.SetBytesUncompressed(n.commitment, true)
-	}
-	return n.cachedPoint
+func (HashedNode) Commit() *Point {
+	panic("can not commit a hash node")
 }
 
-func (n *HashedNode) Commitment() *Point {
-	if n.commitment == nil {
-		panic("nil commitment")
-	}
-	return n.Commit()
+func (HashedNode) Commitment() *Point {
+	panic("can not get commitment of a hash node")
 }
 
-func (*HashedNode) GetProofItems(keylist) (*ProofElements, []byte, [][]byte, error) {
+func (HashedNode) GetProofItems(keylist) (*ProofElements, []byte, [][]byte, error) {
 	return nil, nil, nil, errors.New("can not get the full path, and there is no proof of absence")
 }
 
-func (*HashedNode) Serialize() ([]byte, error) {
+func (HashedNode) Serialize() ([]byte, error) {
 	return nil, errSerializeHashedNode
 }
 
-func (n *HashedNode) Copy() VerkleNode {
-	if n.commitment == nil {
-		panic("nil commitment")
-	}
-	c := &HashedNode{commitment: make([]byte, len(n.commitment))}
-	copy(c.commitment, n.commitment)
-	return c
+func (HashedNode) Copy() VerkleNode {
+	return HashedNode{}
 }
 
-func (n *HashedNode) toDot(parent, path string) string {
-	return fmt.Sprintf("hash%s [label=\"H: %x\"]\n%s -> hash%s\n", path, n.commitment, parent, path)
+func (HashedNode) toDot(parent, path string) string {
+	return fmt.Sprintf("hash%s \n%s -> hash%s\n", path, parent, path)
```

Comment:
This API isn't a big deal since it's for debugging, but we can't print hashes here.

Reply:
I've never used this `toDot` to debug really, but I didn't know has been useful to you.

There's no way to do any resolving since `toDot` doesn't receive a `resolver` interface, so we can only print whatever we have in the parameters which is something I already do here. 

I added a `label="unresolved"` to at least signal that this node exists and is unresolved. If in the future we add a resolver to `toDot`, then we can resolve which is the commitment of this hashed node.